walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,86 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
+  private static final int START_OF_DAY = TimeRange.START_OF_DAY;
+  private static final int END_OF_DAY = TimeRange.END_OF_DAY;
+  private static final TimeRange WHOLE_DAY = TimeRange.WHOLE_DAY;
+  private static final Comparator<TimeRange> ORDER_BY_START = TimeRange.ORDER_BY_START;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // Array of queries representing times to meet.
+    List<TimeRange> query = new ArrayList<>();
+
+    // Copy of events used for iterating and sorting.
+    List<Event> eventsCopy = new ArrayList<>(events);
```

Comment:
This can also be moved lower, right above the call to sort. Doing so saves an allocation if any of the early returns are hit between here and there.

Reply:
Ack